r/R/dplyr-funcs-datetime.R

```diff
@@ -518,6 +518,28 @@ register_bindings_datetime_parsers <- function() {
 
     coalesce_output <- build_expr("coalesce", args = parse_attempt_expressions)
 
-    build_expr("assume_timezone", coalesce_output, options = list(timezone = tz))
+    if (!is.null(tz)) {
+      build_expr("assume_timezone", coalesce_output, options = list(timezone = tz))
+    } else {
+      coalesce_output
+    }
+
   })
+
+  ymd_parser_vec <- c("ymd", "ydm", "mdy", "myd", "dmy", "dym")
+
+  ymd_parser_map_factory <- function(order) {
+    force(order)
```

Comment:
Why do we need to call `force()` here?

Reply:
'cause otherwise it won't work 😬 
https://adv-r.hadley.nz/function-factories.html#forcing-evaluation

Reply:
> 'cause otherwise it won't work

Lol, thanks for that; I think what you're getting at is that it's because it's a function factory. Cool.

Reply:
We need to force evaluation because of otherwise `order` is evaluated lazily.

Reply:
This is the key bit from Advanced R: 
> Unfortunately it doesn’t because x is only evaluated lazily when square() is run, not when power1() is run. In general, this problem will arise whenever a binding changes in between calling the factory function and calling the manufactured function. This is likely to only happen rarely, but when it does, it will lead to a real head-scratcher of a bug.
>
> We can fix this problem by forcing evaluation with [force()](https://rdrr.io/r/base/force.html):

Reply:
I think technically, we can get away without using `force()` here, but it's a bit more robust with it than without it.